src/main/kotlin/com/exactpro/th2/service/generator/protoc/python/PythonFileSpec.kt

```diff
@@ -20,10 +20,16 @@ import com.google.common.base.CaseFormat
 import com.google.common.base.CaseFormat.LOWER_UNDERSCORE
 import java.nio.file.Path
 
-class PythonFileSpec(private val roodDir: Path?, serviceName: String, private val content: String) : FileSpec {
-    private val fileName = CaseFormat.LOWER_CAMEL.to(LOWER_UNDERSCORE, serviceName) + "_service.py"
+class PythonFileSpec(private val roodDir: Path?, pathToFile: String?, serviceName: String, private val content: String) : FileSpec {
 
-    override fun getFilePath(): String = roodDir?.resolve(fileName)?.toString() ?: fileName
+    private val filePath: String
+
+    init {
+        val fileName = CaseFormat.LOWER_CAMEL.to(LOWER_UNDERSCORE, serviceName) + "_service.py"
+        filePath = if (pathToFile != null) Path.of(pathToFile, fileName).toString() else fileName
```

Comment:
You can move it from the init method. Don't see any reason to use it. You can initialize the variable in one line without any problem.
```kotlin
private val filePath: String = "${CaseFormat.LOWER_CAMEL.to(LOWER_UNDERSCORE, serviceName)}_service.py".let {
  if (pathToFile != null) Path.of(pathToFile, it).toString() else it
}
```